Ignore non-replicated parents when load balancing (#2587)
spatialos/UnrealGDK: CHANGELOG.md

- Fixed an issue where possessing a new pawn and immediately setting the owner of the old pawn to the controller could cause server RPCs from that pawn to be dropped.
- Added support for the `bHidden` relevancy flag. Clients will not checkout Actors that have `bHidden` set to true (unless they are always relevant or the root component has collisions enabled).
- Fixed an issue with deployments failing due to the incorrect number of workers when the launch config was specified, rather than automatically generated.

> - Replicated actors with non-replicated owners no longer consider the non-replicated owners for replication.

Reviewer: ```suggestion
- Replicated Actors with non-replicated owners no longer consider the non-replicated owners for replication.
```

- Added support for the `bHidden` relevancy flag. Clients will not checkout Actors that have `bHidden` set to true (unless they are always relevant or the root component has collisions enabled).
- Fixed an issue with deployments failing due to the incorrect number of workers when the launch config was specified, rather than automatically generated.
- Fixed the `too many dynamic subobjects` error on Clients appearing when a Startup Actor, with one dynamic subobject was leaving and re-entering interest multiple times. Added the `RemovedDynamicSubobjectObjectRefs` map in `USpatialPackageMapClient` that keeps the dynamic subobjects removed from Startup Actor's client's interest to avoid duplication of the dynamic subojects when the Startup Actor re-enters the Client's interest.

> - Replicated Actors with non-replicated owners no longer consider the non-replicated owners for replication.

Reviewer 2: By default, only an actor's replicated owners will be used when determining which worker should have authority over an actor. Non-replicated actors are now ignored.

Reviewer 2: Perhaps something like this? ^^

> - Replicated Actors with non-replicated owners no longer consider the non-replicated owners for replication.

Author: ```suggestion
- By default, only an actor's replicated owners will be used when determining which worker should have authority over an actor. Non-replicated actors are now ignored.
```

- Cleaned up startup logs of a few noisy messages.
- Fixed a crash that sometimes occurred upon trying to resolve a pointer to an object that has been unloaded.
- Fixed a crash when spawn requests are forwarded but the `APlayerStart` actor is not resolvable on the target worker.

> - By default, only an actor's replicated owners will be used when determining which worker should have authority over an actor. Non-replicated actors are now ignored.

Reviewer: ```suggestion
- By default, only an Actor's replicated owner hierarchy will be used when determining which worker should have authority over an actor. Non-replicated actors are now ignored.
```
